clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -210,6 +238,12 @@ public String toString() {
             b.append(logStartOffset);
             b.append(", recordErrors: ");
             b.append(recordErrors);
+            b.append(", currentLeader: ");
+            if (currentLeader != null) {
```

Comment:
I _think_ that the following lines could be simply `String.valueOf(currentLeader)`, right?

Reply:
In fact, the `errorMessage` bit could be redone that way too.

Reply:
In fact, I think that the `StringBuilder` code checks for `null`s in its `append()` method.

Reply:
looking at the java docs I think you're right, this could be replaced by `b.append(currentLeader)`. I'm not sure why `errorMessage` was written this way, it looks like it was changed explicitly in [this commit](https://github.com/apache/kafka/commit/f41a5c2c8632bfd0dc50321c1c69418db04f42f6#diff-82aef2b279f7d0093b5e7bbd34cbf9abfa6bb5ed454c72419c03dbe2e58e0eab) but I don't see a reason for it, I could probably change this as well.